opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -116,6 +116,50 @@ namespace Opm {
     }
 
 
+    /// Return true if any well has a THP constraint.
+    template<typename TypeTag>
+    bool
+    BlackoilWellModel<TypeTag>::
+    hasTHPConstraints() const
+    {
+        for (const auto& well : well_container_) {
+            if (well->wellHasTHPConstraints()) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
Did you think about writing this along the lines of
```C++
return std::any_of(well_container.begin(), well_container_.end(),
    [](const Well* well) { return well->wellHasTHPConstraints(); });
```
It's very much in the eye of the beholder, so I'm genuinely asking your opinion here.